plugins/modules/azure_rm_diagnosticsetting.py

```diff
@@ -0,0 +1,575 @@
+                            description:
+                                - A value indicating whether the retention policy is enabled.
+                            type: bool
+                        days:
+                            description:
+                                - The number of days for the retention in days. A value of 0 will retain indefinitely.
+                            type: int
+'''
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+import datetime
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+retention_policy_object = dict(
+    enabled=dict(type='bool', default=True, required=True),
+    days=dict(type='int', default=0, required=True)
+)
+
+log_settings_object = dict(
+    category=dict(type='str', required=True),
+    enabled=dict(type='bool', default=True, required=True),
+    retention_policy=retention_policy_object
+)
+
+metric_settings_object = dict(
+    time_grain=dict(type='str'),
+    category=dict(type='str', required=True),
+    enabled=dict(type='bool', default=True, required=True),
+    retention_policy=retention_policy_object
+)
+
+
+class AzureRMDiagnosticSetting(AzureRMModuleBase):
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            name=dict(type='str', required=True),
+            resource_id=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            workspace_id=dict(type='str'),
+            log_analytics_destination_type=dict(type='str', default=None, choices=['Dedicated', None]),
+            storage_account_id=dict(type='str'),
+            service_bus_rule_id=dict(type='str'),
+            event_hub_name=dict(type='str'),
+            event_hub_authorization_rule_id=dict(type='str'),
+            logs=dict(
+                type='list',
+                elements='dict',
+                options=log_settings_object
+            ),
+            metrics=dict(
+                type='list',
+                elements='dict',
+                options=metric_settings_object
+            )
+        )
+
+        self.name = None
+        self.resource_id = None
+        self.state = None
+        self.workspace_id = None
+        self.log_analytics_destination_type = None
+        self.storage_account_id = None
+        self.service_bus_rule_id = None
+        self.event_hub_name = None
+        self.event_hub_authorization_rule_id = None
+        self.logs = []
+        self.metrics = []
+
+        self.results = dict(changed=False)
+        self.diagnostic_setting_dict = None
+
+        super(AzureRMDiagnosticSetting, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                   supports_check_mode=False,
+                                                   supports_tags=False)
```

Comment:
If you have inherited azure_tags in your document, you cannot set it to False here!
```suggestion
                                                   supports_tags=True)
```